packages/maskbook/src/plugins/RedPacket/SNSAdaptor/hooks/useNftAvailabilityComputed.ts

```diff
@@ -0,0 +1,49 @@
+import { useMemo } from 'react'
+import { compact } from 'lodash-es'
+import { isSameAddress, useChainId, getChainIdFromName, ChainId } from '@masknet/web3-shared'
+import { NftRedPacketJSONPayload, RedPacketStatus } from '../../types'
+import { useAvailabilityNftRedPacket } from './useAvailabilityNftRedPacket'
+
+/**
+ * Fetch the red packet info from the chain
+ * @param payload
+ */
+export function useNftAvailabilityComputed(account: string, payload: NftRedPacketJSONPayload) {
+    const chainId = useChainId()
+    const asyncResult = useAvailabilityNftRedPacket(payload?.rpid, account)
+
+    const result = asyncResult
+    const availability = result.value
+
+    return useMemo(() => {
+        if (!availability) {
+            return {
+                ...asyncResult,
+                payload,
+                computed: {
+                    canClaim: false,
+                    listOfStatus: [] as RedPacketStatus[],
+                },
+            }
+        }
+
+        const isEmpty = availability.balance === '0'
+        const isExpired = availability.expired
+        const isClaimed = availability.isClaimed
+        const isCreator = isSameAddress(payload?.sender.address ?? '', account)
```

Comment:
I feel it's unnecessary to create an `useNftAvailabilityComputed()` hook 🤔, `useAvailabilityNftRedPacket()` is enough.

Reply:
A `NFTRedpacket` can't be refunded